### ISSUE TITLE
Windows version crash

### ISSUE THREAD

**Issue by gamelaster** [gotnospirit/cef3-sdl2#1]
Renderer crash on CefExecuteProcess, any ideas???

**Comment by gotnospirit** [comment]
Hi!
It's been a while since I've done this and I do not have the required setup to test it now but I may help you after the holidays :)
In the meanwhile, can you tell me which versions you have? (windows, visual studio, sdl, cef, cmake)
and is it with this main.cpp or is it with your own code?

**Comment by gamelaster** [comment]
Windows 10 x64, Visual Studio 2017, SDL2 (latest), CEF latest, cmake 3.0.2 (not sure). Building as DEBUG x64 (I changed a bit CMakeLists.txt for x64)
I was testing a clean pulled commit

**Comment by gotnospirit** [comment]
mm, cefbuilds.com seems broken, I get a 403. where did you download it?

**Comment by gamelaster** [comment]
http://opensource.spotify.com/cefbuilds/index.html

**Comment by gotnospirit** [comment]
Thank you!
Unfortunately, using the latest "cef_binary_3.3239.1709.g093cae4_windows64", I end up with compile errors like "SetWindowVisibility is not a member of CefBrowserHost"...
Did wanna try with "cef_binary_3.2704.1414.g185cd6c_windows64" but I was unable to compile itself. seems like a lib link problem...
Anyways, I don't know how I can help you now

**Comment by gamelaster** [comment]
No problem, anyway, I did it working on my private project, I was testing this for see how it's works, maybe I find the problem, since I compiled it successfully, just it crash

**Comment by LinArcX** [comment]
@gamelaster I have same issue. application build successfully but crashes whenever i run it.
Any idea to fix it?

**Comment by gamelaster** [comment]
@LinArcX it was very long time ago. What error do you have?

**Comment by gotnospirit** [comment]
Wow, it's a very old issue.
I need to update my VS in order to try to help you, but can you provide some informations such as the SDL, CMake & CEF versions ?
Thanks

**Comment by LinArcX** [comment]
@gamelaster No error. crashes at runtime.
@gotnospirit 
SDL: 2.0.14
CMAKE: 3.18.4
Windows: 7
Visual Studio: 2019
CEF: 88.2.1+g0b18d0b+chromium-88.0.4324.146

**Comment by LinArcX** [comment]
Oh. I debug the program and realized that this line always return null:
https://github.com/gotnospirit/cef3-sdl2/blob/master/src/main.cpp#L266

**Comment by LinArcX** [comment]
I replaced `SDL_CreateRenderer` with `SDL_CreateSoftwareRenderer`. Now it's not null anymore, but i can't see the homePage(www.google.com).

Edit: I really don't want to use software renderer. If anybody knows how can i solve this issue related to hardware_rendering, please let me know.

**Comment by gotnospirit** [comment]
I've download the lastest SDL (2.0.14) & CEF build (88.2.4+gf3c4ca9+chromium-88.0.4324.150) unfortunately
CEF's API changed since I did this sample
`bool GetViewRect(CefRefPtr<CefBrowser> browser, CefRect &rect)` turned into `void GetViewRect(CefRefPtr<CefBrowser> browser, CefRect &rect)`
and
`SetWindowVisibility` was removed in favor to `WasHidden` alone.
when I try to build libcef_dll_wrapper with cmake, I'm stuck with a `Unknown CMake command "SET_LIBRARY_TARGET_PROPERTIES"`.
I will give it a try later

**Comment by gamelaster** [comment]
@LinArcX if you hw accelerated window is NULL, then something is really wrong. What PC you have? On what are you testing it?

**Comment by LinArcX** [comment]
@gamelaster Sorry. What PC? You mean the version of graphic driver?

**Comment by gamelaster** [comment]
@LinArcX well generally. What GPU do you use, if you have drivers... Don't you run it in VM?

**Comment by LinArcX** [comment]
No. i'm running on real hardware. As i said inside windows 7.And i don't have a dedicated GPU. just intel graphic driver installed.

**Comment by gamelaster** [comment]
@LinArcX please, try to print SDL_GetError if renderer is null to see where is issue.

**Comment by LinArcX** [comment]
I've test it on another machine that has Nvidia graphic driver. It works(the `renderer` isn't null anymore!). but i got a black screen.(There's no error that i can see)

**Comment by Lecrapouille** [comment]
@LinArcX  @gamelaster  @gotnospirit
I succeeded to patch the Linux version https://github.com/Lecrapouille/OffScreenCEF/tree/master/cefsimple_sdl missing some CefSettings to set for CefInitialize (else in my case I got an infinite loop of forks that made my Linux fell down). Feel free to do some pull request for updating the code for Windows. Thanks for your repo. It helped me a lot !
